src/System.Net.Security/src/System/Net/Security/SslState.cs

```diff
@@ -133,11 +133,16 @@ internal void ValidateCreateContext(SslServerAuthenticationOptions sslServerAuth
                 throw new InvalidOperationException(SR.net_auth_client_server);
             }
 
-            if (sslServerAuthenticationOptions.ServerCertificate == null)
+            if (sslServerAuthenticationOptions.ServerCertificate == null && sslServerAuthenticationOptions._serverCertDelegate == null)
```

Comment:
We require a delegate be provided or else we throw?  And if we do throw because ServerCertificate != null but _serverCertDelegate == null, we still use "ServerCertificate" as the exception argument?

Reply:
Hmm, actually, I see why you're doing that.  Makes sense.